jasmine/src/inverted-index.js

```diff
@@ -0,0 +1,149 @@
+class Index {
+
+    // Method to create an inverted-index of the documents
+    createIndex(filePath) {
+
+        // Here, self refers to the Index object
+        var self = this;
+
+        // Initialise Object to store the inverted-index
+        this.indexObj = {};
+
+        /* Fetch the document 
+         * Resolve the promise and store the data in a variable, 'finallySomeData'
+         */
+        return fetch(filePath).then(function(response) {
+            return response.json();
+
+        })
+
+        .then(function(finallySomeData) {
+            self.books = finallySomeData;
+
+            // Call the populateIndex function to create the inverted-index
+            populateIndex();
+
+        });
+
+        function populateIndex() {
+            // Iterate through the data obtaining each book's position in the document.
+            self.books.forEach((book, docIndex) => {
+
+                /* For each document, turn to string, lowercase, remove special characters
+                 * and trim beginning of line spaces.
+                 */
+                var bookObjectString = JSON.stringify(book)
+                    .toLowerCase().replace(/\W/g, ' ')
+                    .replace(/\s+/g, ' ').trim();
+
+                // For each document, split at space to form individual words.
+                var indexArray = bookObjectString.split(' ');
+
+                indexArray.forEach((word, wordIndex) => {
+
+                    /* If the word already exists in the indexObject, 
+                     * add an array with the word's docIndex and wordIndex for this occurrence, 
+                     * under the 'word' property
+                     */
+
+                    if (word in self.indexObj) {
+
+                        self.indexObj[word].push([docIndex, wordIndex]);
+
+                    } else {
+                        // Else create a new array for that word
+                        self.indexObj[word] = [
+                            [docIndex, wordIndex]
+                        ];
+
+                    }
+
+                });
+
+            });
+            //console.log(self.indexObj);
+        }
+
+    }
+
+    // Method to return inverted-index from createIndex method
+    getIndex() {
+        return this.indexObj;
+
+    }
+
+    // Method to search the index for a term
+    searchIndex(term) {
```

Comment:
**Index Functions - Critical**

Your `searchIndex` function can only handle a single string. It should be able to handle 
`index.searchIndex('alice wonderland')` or `index.searchIndex(['alice', 'wonderland', 'qwqwqw')` and return the index for the provided terms or -1 for those that are not found.